xbmc/addons/settings/AddonSettings.cpp

```diff
@@ -438,8 +436,12 @@ bool CAddonSettings::HasSettings() const
 
 bool CAddonSettings::Save()
 {
-  assert(m_addon);
-  return m_addon->SaveSettings();
+  auto addon = m_addon.lock();
```

Comment:
Do you think the usage of auto here is preferred over the shared_ptr usage? It might be trivial for someone experienced with c++ but in my case I had to look into the documentation.
Up to you.

Reply:
I agree. I normally try to use `auto` only if the the type is already stated somewhere else in the expression (e.g. a static_cast) which isn't the case here. Looks like I was feeling lazy today :smile: